source/opt/flatten_array_matrix_stage_var.h

```diff
@@ -0,0 +1,429 @@
+// Copyright (c) 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SOURCE_OPT_FLATTEN_ARRAY_MATRIX_STAGE_VAR_H_
+#define SOURCE_OPT_FLATTEN_ARRAY_MATRIX_STAGE_VAR_H_
+
+#include <unordered_set>
+
+#include "source/opt/pass.h"
+
+namespace spvtools {
+namespace opt {
+
+// A struct for the information of a stage variable's location, component,
```

Comment:
What is a stage variable?  I don't see it anywhere in the spir-v spec.  We should be picking terms that relate to the domain in which we are working.  Using a term like "interface variable" might be better because it matches the name of the list of variable on the [OpEntryPoint](https://www.khronos.org/registry/SPIR-V/specs/unified1/SPIRV.html#OpEntryPoint).